File: app/src/main/java/com/song/deviceinfo/ui/base/BaseAdapter.java
package com.song.deviceinfo.ui.base;

import android.content.Context;

import java.util.List;

import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by chensongsong on 2020/5/25.
 */
public abstract class BaseAdapter<T, K extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<K> {

    protected List<T> data = null;
    protected Context context;

    public BaseAdapter(Context context) {
        this.context = context;
    }

    public void updateData(List<T> data) {
        this.data = data;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        return data == null ? 0 : data.size();
    }

}

File: app/src/main/java/com/song/deviceinfo/ui/base/BaseAdapter.kt
package com.song.deviceinfo.ui.base

import android.content.Context
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by chensongsong on 2020/5/25.
 */
abstract class BaseAdapter<T : Any, VH : RecyclerView.ViewHolder>(
    protected val context: Context
) : RecyclerView.Adapter<VH>() {
    
    protected val items = mutableListOf<T>()
    
    fun setData(newItems: List<T>) {
        items.clear()
        items.addAll(newItems)
        notifyDataSetChanged()
    }
    
    override fun getItemCount(): Int = items.size
}
